cmd/clusterctl/client/repository/components_client.go

```diff
@@ -47,18 +48,53 @@ func newComponentsClient(provider config.Provider, repository Repository, config
 	}
 }
 
+func (f *componentsClient) GetWithValues(version, targetNamespace, watchingNamespace string) (Components, error) {
```

Comment:
Suggestion: 
```suggestion
type ComponentsClientInput struct {
  version, targetNamespace, watchingNamespace string
  withValues bool // Feel free to choose a more appropriate name
}
func (f *componentsClient) Get(input ComponentsClientInput) (Components, error) {
}
```
Then we can pass in the `withValues` to `NewComponents`.

Reply:
I also noticed that we are missing the godoc for these methods. Maybe we can add them in here now.

Reply:
will add godoc